---
title: 'SyntaxError: return not in function'
slug: Web/JavaScript/Reference/Errors/Bad_return_or_yield
tags:
  - Errors
  - JavaScript
  - SyntaxError
translation_of: Web/JavaScript/Reference/Errors/Bad_return_or_yield
---
<div>{{jsSidebar("Errors")}}</div>

<h2 id="メッセージ">メッセージ</h2>

<pre class="syntaxbox">SyntaxError: return not in function
SyntaxError: yield not in function
</pre>

<h2 id="エラータイプ">エラータイプ</h2>

<p>{{jsxref("SyntaxError")}}。</p>

<h2 id="何がうまくいかなかったのか？">何がうまくいかなかったのか？</h2>

<p><code><a href="/ja/docs/Web/JavaScript/Reference/Statements/return">return</a></code> ステートメントか <code><a href="/ja/docs/Web/JavaScript/Reference/Operators/yield">yield</a></code> ステートメントが、 <a href="/ja/docs/Web/JavaScript/Guide/Functions">関数</a> の外側で呼び出されました。どこかで、波括弧を忘れたのかもしれません。<code>return</code> ステートメントと <code>yield</code> ステートメントは、関数内で使用しなければなりません。なぜなら、これらのステートメントは、関数の実行を終了（または、停止や再開）し、関数の呼び出し元に返す値を指定するからです。</p>

<h2 id="例">例</h2>

<pre class="brush: js example-bad">var cheer = function(score) {
  if (score === 147)
    return "Maximum!";
  };
  if (score &gt; 100) {
    return "Century!";
  }
}

// SyntaxError: return not in function</pre>

<p>一見して、波括弧は正しく見えますが、このコードスニペットでは、最初の <code>if</code> ステートメントの後の <code>{</code> を忘れています。正しくは以下のようにします：</p>

<pre class="brush: js example-good">var cheer = function(score) {
  if (score === 147) {
    return "Maximum!";
  }
  if (score &gt; 100) {
    return "Century!";
  }
};</pre>

<h2 id="関連項目">関連項目</h2>

<ul>
 <li><code><a href="/ja/docs/Web/JavaScript/Reference/Statements/return">return</a></code></li>
 <li><code><a href="/ja/docs/Web/JavaScript/Reference/Operators/yield">yield</a></code></li>
</ul>
